### PR TITLE
PIL-168:Task 3: enter group's accounting period

### PR DESCRIPTION
**Pull request by nitin-srivastav**

### CODE REVIEW

File: app/controllers/subscription/MneOrDomesticController.scala [hunk 1]
```diff
@@ -71,13 +70,32 @@ class MneOrDomesticController @Inject() (
...
               Future
-                .fromTry(request.userAnswers.set(SubscriptionPage, Subscription(domesticOrMne = value, subscriptionStatus = RowStatus.InProgress)))
+                .fromTry(
+                  request.userAnswers.set(
+                    SubscriptionPage,
+                    Subscription(
+                      domesticOrMne = value,
+                      groupDetailStatus = RowStatus.InProgress,
+                      accountingPeriod = subscriptionData.accountingPeriod,
+                      primaryContactName = subscriptionData.primaryContactName,
```
Comment (Al-kouhestani) [inline app/controllers/subscription/MneOrDomesticController.scala:86]: We don't need all these values, they are defaulted to none so you really only need to enter a value for those fields needed i.e. domesticorMne and status. I am not sure why you have edited this controller

File: app/viewmodels/checkAnswers/GroupAccountingPeriodSummary.scala [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+      val startDate = HtmlFormat.escape(answer.startDate.toString) + "<br>"
+      val endDate   = HtmlFormat.escape(answer.endDate.toString) + "<br>"
+      val value     = startDate + endDate
+
+      SummaryListRowViewModel(
+        key = "groupAccountingPeriod.checkYourAnswersLabel",
+        value = ValueViewModel(HtmlContent(value)),
+        actions = Seq(
+          ActionItemViewModel("site.change", controllers.subscription.routes.GroupAccountingPeriodController.onPageLoad(CheckMode).url)
+            .withVisuallyHiddenText(messages("groupAccountingPeriod.change.hidden"))
```
Comment (Al-kouhestani) [inline app/viewmodels/checkAnswers/GroupAccountingPeriodSummary.scala:41]: This is not the right hidden message

File: test/controllers/GroupAccountingPeriodControllerSpec.scala [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+        val view = application.injector.instanceOf[GroupAccountingPeriodView]
+
+        val result = route(application, request).value
+
+        status(result) mustEqual BAD_REQUEST
+        contentAsString(result) mustEqual view(boundForm, NormalMode)(request, appConfig(application), messages(application)).toString
+      }
+    }
+
```
Comment (Al-kouhestani) [inline test/controllers/GroupAccountingPeriodControllerSpec.scala:98]: Please add an additional unit test for remapFormErrors method in the controller

**Review by Al-kouhestani — CHANGES_REQUESTED** [review]

**Comment by nitin-srivastav** [comment]
The changes have been merged with another PR PIL-168-Al

**Pull request closed without merging**